src/renderer/pages/Operations/components/ActionSteps/Confirmation.tsx

```diff
@@ -1,22 +1,51 @@
-import { MultisigTransaction, Transaction, Fee } from '@renderer/entities/transaction';
+import { useEffect, useState } from 'react';
+import { useStore } from 'effector-react';
+
+import {
+  MultisigTransaction,
+  Transaction,
+  Fee,
+  DepositWithLabel,
+  isXcmTransaction,
+  TransactionType,
+  XcmFee,
+} from '@renderer/entities/transaction';
 import { TransactionAmount } from '@renderer/pages/Operations/components/TransactionAmount';
-import { DetailRow, FootnoteText, Icon } from '@renderer/shared/ui';
+import { Button, DetailRow, FootnoteText, Icon } from '@renderer/shared/ui';
 import { ExtendedChain } from '@renderer/entities/network';
 import { useI18n } from '@renderer/app/providers';
 import { getIconName } from '../../common/utils';
-import type { MultisigAccount } from '@renderer/shared/core';
+import type { Account, MultisigAccount } from '@renderer/shared/core';
 import Details from '../Details';
+import { sendAssetModel } from '@renderer/widgets';
+import { getAssetById } from '@renderer/shared/lib/utils';
 
 type Props = {
   tx: MultisigTransaction;
   account: MultisigAccount;
+  signatory?: Account;
   connection: ExtendedChain;
   feeTx?: Transaction;
+  onSign: () => void;
 };
-export const Confirmation = ({ tx, account, connection, feeTx }: Props) => {
+export const Confirmation = ({ tx, account, connection, signatory, feeTx, onSign }: Props) => {
+  const xcmConfig = useStore(sendAssetModel.$finalConfig);
   const { t } = useI18n();
+  const [isFeeLoaded, setIsFeeLoaded] = useState(false);
+
+  const asset = getAssetById(tx.transaction?.args.assetId, connection.assets) || connection.assets[0];
 
   const iconName = getIconName(tx.transaction);
+  const transaction =
+    tx.transaction?.type === 'batchAll'
+      ? tx.transaction.args.transactions.find(
+          (tx: Transaction) => tx.type === TransactionType.BOND || tx.type === TransactionType.UNSTAKE,
+        ) || tx.transaction.args.transactions[0]
+      : tx.transaction;
```

Comment:
maybe make some improvements for this code? Move it to function (this code uses in few places), create variable for find value, use array.includes for type detection